ledger/db/src/lib.rs

```diff
@@ -369,67 +341,57 @@ impl LedgerDB {
         Ok(())
     }
 
-    fn write_transactions_by_block(
-        &self,
-        block_index: u64,
-        transactions: &[RedactedTx],
-        db_transaction: &mut RwTransaction,
-    ) -> Result<(), lmdb::Error> {
-        db_transaction.put(
-            self.transactions_by_block,
-            &u64_to_key_bytes(block_index),
-            &serialize(&transactions.to_vec())
-                .unwrap_or_else(|_| panic!("Could not serialize chunk {:?}", transactions)),
-            WriteFlags::empty(),
-        )?;
-        Ok(())
-    }
-
-    /// Checks if a block can be appended to the db
+    /// Checks if a block can be appended to the db.
     fn validate_append_block(
         &self,
         block: &Block,
-        transactions: &[RedactedTx],
+        block_contents: &BlockContents,
     ) -> Result<(), Error> {
-        // We don't want empty blocks
-        if transactions.is_empty() {
-            return Err(Error::NoTransactions);
-        }
-
         // Check that version is correct
         if block.version != BLOCK_VERSION {
             return Err(Error::InvalidBlock);
         }
 
-        // Check if block is being appended at the correct place
+        // A block must have outputs.
+        if block_contents.outputs.is_empty() {
+            // TODO: better error type.
+            return Err(Error::InvalidBlock);
+        }
+
+        // TODO: enable this.
```

Comment:
I'd like to add this new check. It breaks several unit tests, so I'd like to deal with it as a follow-on PR,
https://mobilecoin.atlassian.net/browse/MC-1335